src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -981,9 +981,7 @@ public function createService(Definition $definition, $id, $tryProxy = true)
         }
 
         if ($callable = $definition->getConfigurator()) {
-            if (is_array($callable)) {
-                $callable[0] = $callable[0] instanceof Reference ? $this->get((string) $callable[0]) : $parameterBag->resolveValue($callable[0]);
-            }
+            $callable = $this->resolveServices($parameterBag->resolveValue($callable));
```

Comment:
well, we don't need to support the case of arrays either for configurators first element. So this is just a matter of adding the support of Definition object in the current code (which already supports references)